src/compiler/crystal/command.cr

```diff
@@ -511,9 +520,30 @@ class Crystal::Command
       @color = false
       compiler.color = false
     end
+    setup_compiler_warning_options(opts, compiler)
     opts.invalid_option { }
   end
 
+  private def setup_compiler_warning_options(opts, compiler)
+    compiler.warnings_exclude << Crystal.normalize_path "lib"
+    opts.on("--warnings emit|error|ignore", "How to treat warnings. (default: emit)") do |w|
+      compiler.warnings = case w
+                          when "emit"
```

Comment:
We should probably not try to invent the wheel. I think everywhere else in the world there are two options: "warnings on or off", and "treat warnings as errors".

Reply:
You're maybe right. I usually associate `emit` as the output format (object file, LLVM IR, JSON...).

Reply:
Ok, do we prefer to start with `--warnings all|none` and `--error-on-wargnings` in case the 
warnings categories increase in the future so we can have `--warnings deprecated,X,Y,Z` to specify which warnings we want?

Or seems enough to go with `--warnings warn|error|ignore`?

Reply:
Having warning categories is great. For instance on `pylint`, there is a `-d, --disable=` to disable messages/reports. I think this will be appropriate to copy this opt-in by default approach, for example having `disable-warnings` , and only disable warnings we don't want.

There is also a `disable=all` to disable all messages/reports, and `enable` to enable some.
For example `disable=all enable=classes`

Reply:
I think `--warnings all|none` with `--error-on-warnings` is fine. Then we could eventually list specific warnings, as you suggest.